components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2181,6 +2185,56 @@ public ServiceProvider getApplication(int applicationId) throws IdentityApplicat
         }
     }
 
+    private List<ServiceProviderProperty> removeAndSetSpProperties(List<ServiceProviderProperty> propertyList) {
+
+        /* These properties are either first class or part of advanced configurations and hence removing
+        them as they can't be packed as a part of additional sp properties again.*/
+        List<ServiceProviderProperty> copiedProps = new ArrayList<>();
+        for (ServiceProviderProperty serviceProviderProp: propertyList) {
+            // copy, remove and then set, not to tangle with if requested properties have any of these specifically.
+            copiedProps.add(serviceProviderProp);
+        }
+        copiedProps.removeIf(property -> SKIP_CONSENT.equals(property.getName()));
+        copiedProps.removeIf(property -> SKIP_LOGOUT_CONSENT.equals(property.getName()));
+        copiedProps.removeIf(property -> USE_DOMAIN_IN_ROLES.equals(property.getName()));
+        copiedProps.removeIf(property -> USE_USER_ID_FOR_DEFAULT_SUBJECT.equals(property.getName()));
+        copiedProps.removeIf(property -> TEMPLATE_ID_SP_PROPERTY_NAME.equals(property.getName()));
+        copiedProps.removeIf(property -> IS_MANAGEMENT_APP_SP_PROPERTY_NAME.equals(property.getName()));
+        return copiedProps;
+    }
```

Comment:
Let's not remove these as it will have undesired effects on the SOAP clients. If needed let's try to remove these just from the REST API

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/4065/commits/80d85b77d456dcccfac5e78d7446eb57dd687eb5